integrationtesting/opendcs-tests/src/test/java/org/opendcs/dao/CwmsLocationLevelDAOTestIT.java

```diff
@@ -0,0 +1,584 @@
+package org.opendcs.dao;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assumptions.*;
+
+import java.nio.charset.StandardCharsets;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.opendcs.database.api.OpenDcsDatabase;
+import org.opendcs.fixtures.AppTestBase;
+import org.opendcs.fixtures.annotations.ConfiguredField;
+
+import decodes.cwms.CwmsTimeSeriesDb;
+import decodes.cwms.CwmsLocationLevelDAO;
+import decodes.tsdb.TimeSeriesDb;
+import org.opendcs.database.dai.SiteReferenceMetaData;
+
+import org.opendcs.model.SiteReferenceValue;
+import org.opendcs.model.SiteReferenceSpecification;
+import org.opendcs.model.cwms.CwmsSiteReferenceValue;
+import org.opendcs.model.cwms.CwmsSiteReferenceSpecification;
+
+import org.opendcs.fixtures.annotations.EnableIfTsDb;
+import org.opendcs.utils.logging.OpenDcsLoggerFactory;
+import org.slf4j.Logger;
+import java.util.Date;
+import java.util.Calendar;
+import decodes.tsdb.CTimeSeries;
+import ilex.var.TimedVariable;
+
+/**
+ * Integration tests for CwmsLocationLevelDAO
+ * These tests require a real CWMS database connection
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class CwmsLocationLevelDAOTestIT extends AppTestBase
+{
+    private static final Logger log = OpenDcsLoggerFactory.getLogger();
+
+    @ConfiguredField
+    private TimeSeriesDb tsDb;
+    
+    @ConfiguredField
+    private OpenDcsDatabase db;
+
+    private CwmsLocationLevelDAO dao;
+    private static boolean testDataLoaded = false;
+    
+    @BeforeAll
+    public void setupTestData() throws Exception
+    {
+        if (tsDb instanceof CwmsTimeSeriesDb && !testDataLoaded)
+        {
+            try
+            {
+                log.info("Loading location level test data for integration tests");
+                loadLocationLevelTestData();
+                testDataLoaded = true;
+                log.info("Location level test data loaded successfully");
+            }
+            catch (Exception ex)
+            {
+                log.error("Failed to load test data: " + ex.getMessage(), ex);
+                throw ex;
+            }
+        }
+    }
+    
+    @BeforeEach
+    public void setUp() throws Exception
+    {
+        if (tsDb instanceof CwmsTimeSeriesDb)
+        {
+            Optional<SiteReferenceMetaData>  dai = db.getDao(SiteReferenceMetaData.class);
```

Comment:
You are stuck creating a "makeSiteReferenceMetaDataDao" method in the DatabaseConnectionOwner interface.